v2/client.go

```diff
@@ -285,7 +285,18 @@ func (c *Client) buildRequest(ctx context.Context, method string, uri *url.URL,
 	return req, nil
 }
 
-func (c *Client) do(req *http.Request, out interface{}) error {
+// body wraps do (typically (*Client).do), passing a *T to do, and returning
+// exactly one non-zero value depending on the result of do.
+func body[T any](do func(_ *http.Request, out any) error, req *http.Request) (*T, error) {
```

Comment:
It's a shame that Go doesn't allow type parameters on methods, else we could have something like `client.DoForBody[T]()`. Yours is a good workaround, plus I always enjoy higher order functions.